VMBackup/main/Utils/SizeCalculation.py

```diff
@@ -60,32 +111,153 @@ def get_loop_devices(self):
+                    logicalVolumeGroup = pvs_item_split[1]
+                    if(logicalVolumeGroup in self.pvs_dict.keys()):
+                        self.pvs_dict.get(logicalVolumeGroup).append(physicalVolume)
+                    else:
+                        self.pvs_dict[logicalVolumeGroup] = []
+                        self.pvs_dict.get(logicalVolumeGroup).append(physicalVolume)
+            self.logger.log("The pvs_dict contains {0}".format(str(self.pvs_dict)))
+        except Exception as e:
+            errMsg = 'Failed to serialize pvs_output with error: %s, stack trace: %s' % (str(e), traceback.format_exc())
+            self.logger.log(errMsg, True, 'Error') 
+        for lvg in self.pvs_dict.keys():
+            count = 0
+            for disk in self.disksToBeIncluded:
+                for pv in self.pvs_dict[lvg]:
+                    if(disk in pv):
+                        count = count+1
+            if(count == len(self.pvs_dict[lvg])):
+                lvg = "/dev/mapper/" + lvg
+                self.logicalVolume_to_bill.append(lvg)
+            else:
+                self.logger.log("Partial snapshotting for the logical volume group {0} can't be taken".format(lvg))
+        self.logger.log("the lvm list to bill are {0}".format(self.logicalVolume_to_bill))
+
     def device_list_for_billing(self):
         self.logger.log("In device_list_for_billing",True)
         devices_to_bill = [] #list to store device names to be billed
         device_items = disk_util.get_device_items(None)
         for device_item in device_items :
-            # self.logger.log("Device name : {0} ".format(str(device_item.name)),True)
             if str(device_item.name).startswith("sd"):
                 devices_to_bill.append("/dev/{0}".format(str(device_item.name)))
+            else:
+                self.logger.log("Not adding device {0} as it does not start with sd".format(str(device_item.name)))
+        self.logger.log("Initial billing items {0}".format(devices_to_bill))
+        
+        '''
+            Sample output for file_systems_info
+            [('sysfs', 'sysfs', '/sys'), ('proc', 'proc', '/proc'), ('udev', 'devtmpfs', '/dev'),..]
+            Since root devices are at mount points '/' and '/boot/efi' we use file_system_info to find the root_devices based on the mount points.
+        '''
+
+        for file_system in self.file_systems_info:
+            if(file_system[2] in self.root_mount_points):
+                self.root_devices.append(file_system[0])
+        self.logger.log("root_devices {0}".format(str(self.root_devices)))
+        self.logger.log("lsscsi_list {0}".format(self.lsscsi_list))
+        '''
+            Sample output of the lsscsi command 
+            [1:0:0:15]   disk    Msft     Virtual Disk     1.0   /dev/sda
+            [1:0:0:18]   disk    Msft     Virtual Disk     1.0   /dev/sdc
+        '''
+
+        self.disksToBeIncluded = self.disk_list_for_billing() 
```

Comment:
this pattern seems inconsistent. In disk_list_for_billing, we utilize the returned value even though the returned value itself is self.disksToBeIncluded and we do not return any value in get_logicalVolumes_for_billing method.